server/re/src/icatGeneralMS.cpp

```diff
@@ -138,11 +138,14 @@ msiQuota( ruleExecInfo_t *rei ) {
  * \sa none
 **/
 int  msiSetResource( msParam_t* xrescName, ruleExecInfo_t *rei ) {
-    char *rescName;
+    char* rescName{static_cast<char *>(xrescName->inOutStruct)};
 
-    rescName = ( char * ) xrescName->inOutStruct;
+    if (rei->doi == nullptr) {
```

Comment:
Swap the operands for extra protection.

Reply:
I'm assuming this is for preventing accidental `=` instead of `==`, right?

Reply:
Correct.